diff --git a/sympy/physics/control/control_plots.py b/sympy/physics/control/control_plots.py
index f9f1fcdd0e..e3ece9286d 100644
--- a/sympy/physics/control/control_plots.py
+++ b/sympy/physics/control/control_plots.py
@@ -1,5 +1,7 @@
 from __future__ import annotations
 from sympy.core.numbers import I, pi
+from sympy.functions.elementary.miscellaneous import sqrt
+from sympy.functions.elementary.complexes import re, im
 from sympy.functions.elementary.exponential import (exp, log)
 from sympy.polys.partfrac import apart
 from sympy.core.symbol import Dummy
@@ -748,6 +750,16 @@ def bode_magnitude_numerical_data(system, initial_exp=-5, final_exp=5, freq_unit
         repl = I*_w
     w_expr = expr.subs({system.var: repl})
 
+    # NOTE: instead of Abs(w_expr), another way would be
+    # sqrt(re(w_expr)**2 + im(w_expr)**2)
+    # but re(), im() are going to evaluate the symbolic expression, potentially
+    # adding a significant overhead for longer transfer functions.
+    mag = 20*log(Abs(w_expr), 10)
+
+    # We are computing the magnitude, which must return real data.
+    # However, numerical computation might introduce imaginary parts.
+    # Let's request the data series to discard any imaginary part.
+    kwargs["return"] = "real"
     mag = 20*log(Abs(w_expr), 10)
 
     x, y = LineOver1DRangeSeries(mag,
